pkgs/top-level/all-packages.nix

```
@@ -14870,7 +14870,7 @@ with pkgs;
   lttv = callPackage ../development/tools/misc/lttv { };
 
   luaformatter = callPackage ../development/tools/luaformatter
-    (lib.optionalAttrs stdenv.isDarwin {
+    (lib.optionalAttrs (stdenv.hostPlatform.system == "x86_64-darwin") {
```

```suggestion
    (lib.optionalAttrs (stdenv.cc.isClang && lib.versionOlder stdenv.cc.version "9") {
```

Would this be a little more accurate, as to the reason the compiler needs to be overridden?

---

We have split the behaviour between x64_64 and aarch64 darwin. For unknown reason the aarch64 needs to be compiled with the default stdenv and x86_64 with clang9 stdenv. The current proposal should fit perfectly.

---

> For unknown reason the aarch64 needs to be compiled with the default stdenv and x86_64 with clang9 stdenv

I think this is because aarch64 already uses llvm 11, as it is required for apple silicon support, so overriding with clang 9 unconditionally would downgrade it.

> The current proposal should fit perfectly.

Agreed, and my suggestion (currently) should have the same effect. I'm hoping that this predicate will better reflect the underlying cause.

---

Thanks to your explanation I understand the underling problem and I think you are right. If you want to build luaformatter with clang, you have to have at least clang9. Here, the system is unrelated. I applied your proposal.